pkg/systemd/hwinfo.jsx

```diff
@@ -75,16 +75,62 @@ class HardwareInfo extends React.Component {
 
     render() {
         let pci = null;
+        let memory = null;
+        let $ = require("jquery");
 
         if (this.props.info.pci.length > 0) {
             let sortedPci = this.props.info.pci.concat();
             sortedPci.sort((a, b) => a[this.state.sortBy].localeCompare(b[this.state.sortBy]));
 
             pci = (
-                <Listing title={ _("PCI") } columnTitles={ [ _("Class"), _("Model"), _("Vendor"), _("Slot") ] }
-                         columnTitleClick={ index => this.setState({ sortBy: this.sortColumnFields[index] }) } >
-                    { sortedPci.map(dev => <ListingRow key={dev.slot} columns={[ dev.cls, dev.model, dev.vendor, dev.slot ]} />) }
-                </Listing>
+                <div id="pci_table">
+                    <Listing title={ _("PCI") } columnTitles={ [ _("Class"), _("Model"), _("Vendor"), _("Slot") ] }
+                             columnTitleClick={ index => this.setState({ sortBy: this.sortColumnFields[index] }) } >
+                        { sortedPci.map(dev => <ListingRow columns={[ dev.cls, dev.model, dev.vendor, dev.slot ]} />) }
+                    </Listing>
+                </div>
+            );
+        }
+
+        if (this.props.info.memory.array.length > 0) {
+            let empty_span = null;
+            let display_all = function(e) {
+                $('#memory_table').addClass('show-all-slots');
+                $('#view-all-slots').hide();
+            };
+            let empty_slots = this.props.info.memory.empty_slots;
+            if (this.props.info.memory.empty_slots > 0) {
+                empty_span = (
+                    <span className="ct-hardware-memory-empty-count">
+                        {empty_slots} empty slots
+                        <a
+                            href="#memory_table"
+                            id="view-all-slots"
+                            onClick={display_all}
+                        >
+                            view all
+                        </a>
+                    </span>
+                );
+            }
+            memory = (
+                <div id="memory_table">
+                    <Listing title={ _("Memory") } actions={ [ empty_span ] }
+                             columnTitles={ [ _("ID"), _("Description"), _("Vendor"), _("Model"), _("Size"), _("Clock Speed"), _("Serial") ] } >
+                        { this.props.info.memory.array.map(dimm => {
+                            var list = null;
+                            if (dimm.type_detail == "None") {
+                                empty_slots += 1;
```

Comment:
This is never being used. If it was supposed to be a row counter for showing in the table, you would need to start at 1, not at `this.props.info.memory.empty_slots`.